ql/src/test/org/apache/hadoop/hive/ql/udf/generic/TestGenericUDFFromUnixTime.java

```diff
@@ -0,0 +1,132 @@
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+import org.apache.hadoop.mapred.JobConf;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.DateTimeFormatterBuilder;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * TestGenericUDFFromUnixTime.
+ */
+public class TestGenericUDFFromUnixTime {
+
+  public static void runAndVerify(GenericUDFFromUnixTime udf,
+      Object arg, Object expected) throws HiveException {
+    DeferredObject[] args = { new DeferredJavaObject(arg) };
+    Object result = udf.evaluate(args);
+    if (expected == null) {
+      assertNull(result);
+    } else {
+      assertEquals(expected.toString(), result.toString());
+    }
+  }
+
+  public static void runAndVerify(GenericUDFFromUnixTime udf,
+      Object arg1, Object arg2, Object expected) throws HiveException {
+    DeferredObject[] args = { new DeferredJavaObject(arg1), new DeferredJavaObject(arg2) };
+    Object result = udf.evaluate(args);
+
+    if (expected == null) {
+      assertNull(result);
+    } else {
+      assertEquals(expected.toString(), result.toString());
+    }
+  }
+
+  @Test
+  public void testTimestampDefaultTimezone() throws HiveException {
+    ObjectInspector valueLongOI = PrimitiveObjectInspectorFactory.writableLongObjectInspector;
+    GenericUDFFromUnixTime udf = new GenericUDFFromUnixTime();
+    ObjectInspector args[] = {valueLongOI};
+    udf.initialize(args);
+
+    Timestamp ts = Timestamp.valueOf("1470-01-01 00:00:00");
+    TimestampTZ tstz = TimestampTZUtil.convert(ts, ZoneId.systemDefault());
+
+    runAndVerify(udf,
+            new LongWritable(tstz.getEpochSecond()), new Text("1470-01-01 00:00:00"));
+
+    // test null values
+    runAndVerify(udf, null, null);
+  }
+
+  @Test
+  public void testTimestampOtherTimezone() throws HiveException {
+    ObjectInspector valueLongOI = PrimitiveObjectInspectorFactory.writableLongObjectInspector;
+    GenericUDFFromUnixTime udf = new GenericUDFFromUnixTime();
+    ObjectInspector args[] = {valueLongOI};
+    udf.initialize(args);
+
+    Timestamp ts = Timestamp.valueOf("2010-01-13 11:57:40");
+    TimestampTZ tstz1 = TimestampTZUtil.convert(ts, ZoneId.of("America/Los_Angeles"));
+    TimestampTZ tstz2 = TimestampTZUtil.convert(ts, ZoneId.of("America/New_York"));
+    TimestampTZ tstz3 = TimestampTZUtil.convert(ts, ZoneId.of("Europe/London"));
+
+    runAndVerify(udf,
+            new LongWritable(tstz1.getEpochSecond()), new Text("2010-01-13 11:57:40"));
+    runAndVerify(udf,
+            new LongWritable(tstz2.getEpochSecond()), new Text("2010-01-13 08:57:40"));
+    runAndVerify(udf,
+            new LongWritable(tstz3.getEpochSecond()), new Text("2010-01-13 03:57:40"));
+  }
+
+  @Test
+  public void testTimestampWithArg2() throws HiveException {
+    ObjectInspector valueLongOI = PrimitiveObjectInspectorFactory.writableLongObjectInspector;
```

Comment:
Can we have same test with PrimitiveObjectInspectorFactory.writableIntObjectInspector as well?

Reply:
IntObjectInspector is removed from the code. so no need to add anything for that in the test.